src/Presentation/FacetUiBuilder.php

```diff
@@ -109,25 +117,50 @@ private function getExampleListItems(): array {
 	}
 
 	/**
-	 * @param array<array<string, mixed>> $items
+	 * @return array<array<string, mixed>>
 	 */
-	private function getListFacetHtml( array $items ): string {
-		return $this->parser->processTemplate(
-			'ListFacet',
-			[ 'items' => $items ]
-		);
-	}
-
-	private function getRangeFacetHtml( ?int $currentMin = null, ?int $currentMax = null ): string {
-		return $this->parser->processTemplate(
-			'RangeFacet',
+	private function getExampleRangeItems(): array {
+		return [
 			[
+				'type' => 'Range',
 				'msg-min' => wfMessage( 'wikibase-faceted-search-facet-range-min' )->text(),
 				'msg-max' => wfMessage( 'wikibase-faceted-search-facet-range-max' )->text(),
-				'current-min' => $currentMin,
-				'current-max' => $currentMax,
+				'current-min' => 1900,
+				'current-max' => 2024,
+			],
+			[
+				'type' => 'Range',
+				'msg-min' => wfMessage( 'wikibase-faceted-search-facet-range-min' )->text(),
+				'msg-max' => wfMessage( 'wikibase-faceted-search-facet-range-max' )->text(),
+				'current-min' => 10
 			]
-		);
+		];
+	}
+
+	/**
+	 * @param array<array<string, mixed>> $items
+	 */
+	private function getItemsHtml( array $items ): string {
+		if ( $items === [] ) {
+			return '';
+		}
+
+		$html = '';
+		foreach ( $items as $item ) {
+			if (
+				!isset( $item['type'] ) ||
+				!is_string( $item['type'] ) ||
+				!strlen( $item['type'] ) > 0
+			) {
+				continue;
+			}
+
+			$html .= $this->parser->processTemplate(
```

Comment:
Would be cleaner if you do

```
if ( $this->hasValidType( $item ) ) {
    $html .= $this->parser->processTemplate();
}
```

Reply:
(Sidenote: I suspect that once we finish up the code, we will not have an $items array any more, and have more guarantees about the data we get. Bit hard to see as this class is very WIP)